### ISSUE TITLE
`math avg` and `math stddev` do not accept the same inputs

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#10632]
### Describe the bug

as in the title, the issue being mostly with `duration`s and `filesize`s.

### How to reproduce

1. define a few `duration`s
```nushell
let random_durations = 1..10 | each { (random int) mod 100 } | into duration
```
2. compute their mean
```nushell
> $random_durations | math avg
54ns
```
3. try to compute their standard deviation
```nushell
> $random_durations | math stddev
Error: nu::shell::unsupported_input

  × Unsupported input
   ╭─[entry #14:1:1]
 1 │ let random_durations = 1..10 | each { (random int) mod 100 } | into duration
   ·                                                                ──────┬──────
   ·                                                                      ╰── input type: value originates from here
   ╰────
   ╭─[entry #16:1:1]
 1 │ $random_durations | math stddev
   · ────────┬────────
   ·         ╰── Attempted to compute the sum of squares of a non-int, non-float value
   ╰────
```
4. try again by replacing all "duration"s by "filesize" and you'll get the same exact behaviour

### Expected behavior

because the mean and the standard deviation are two tools on data sequences that are as common, i expected them to accept the same inputs :yum: 

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.85.1                                   |
| branch             | main                                     |
| commit_hash        | 8c507dc984693c45b0440adf63a94e0410b4bc42 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31)      |
| rust_channel       | 1.70.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.70.0 (ec8a8a0ca 2023-04-25)      |
| build_time         | 2023-10-04 18:35:49 +02:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | gstat, nu_plugin_explore                 |

**Comment by fdncred** [comment]
I went into pretty good on your last issue about this. Not sure where that's at.

**Comment by amtoine** [comment]
you mean https://github.com/nushell/nushell/issues/10617?

it looks related but rather two sides of a similar issue, right?
- https://github.com/nushell/nushell/issues/10617 is more about the `duration`s being in a `table` or not
- here it's more about whether or not `math stddev` accepts `duration`s

i might be confused about whether or not these two are the same :cry:

**Comment by fdncred** [comment]
yup, #10617. Sorry I didn't link it before. Agreed, they're related. 

Generally speaking, if you look at the input/outputs, I 100% agree that they're different and when I saw that when I was researching 10617 I immediately thought that we need to look at a lot of commands that do similar things, like more of the math commands (and perhaps others), and make sure they all have similar i/o types. Also, to having similar i/o types, we should make sure they're in the same order with table coming last (until the i/o matching bug is fixed).

This is probably the order I'd put them in since JT confirmed that they are tried top to bottom.
```rust
                (Type::Int, Type::Float),
                (Type::String, Type::Float),
                (Type::Bool, Type::Float),
                (Type::Float, Type::Float),
                (
                    Type::List(Box::new(Type::Any)),
                    Type::List(Box::new(Type::Float)),
                ),
                (Type::Record(vec![]), Type::Record(vec![])),
                (Type::Table(vec![]), Type::Table(vec![])),
```

**Comment by amtoine** [comment]
ooh yeah, very helpful information :+1: 
will see if i have time for these two at least :crossed_fingers:

**Comment by sholderbach** [comment]
I think the initial motivation why the core of `math stddev` (i.e. variance) does not accept numerics with units makes some sense, but it breaks down for `math stddev`.

https://github.com/nushell/nushell/blob/9f144798d3f8244cbc324236dd4c546f749ec9d4/crates/nu-command/src/math/stddev.rs#L68-L79

For `math variance` the value would be the square of the unit and not the unit. We can not cleanly represent that as we don't have a Units of measurement system included with our types.

But as the standard deviation has the same unit as the input we should be able to support that. Doing so would require a bit of work to lift either the checking to a different place or define the standard deviation operation differently.
